File: root/app/.api/messages-routes.ts
import { NextResponse } from 'next/server';
import { dbService } from '../.db/db-service';

export async function GET(req: Request): Promise<NextResponse> {
    const { searchParams } = new URL(req.url);
    const chatId = searchParams.get('chatId');
    if(!chatId) return NextResponse.json({ error: 'chatId is required' }, { status: 400 });

    try {
        const messages = await dbService.messagesConfig.getMessagesByChatId(chatId);
        return NextResponse.json(messages);
    } catch(err) {
        return NextResponse.json({ error: 'Failed to load messages' }, { status: 500 });
    }
}

File: main/src/main/java/com/app/main/root/app/_api/messages-routes.py
from fastapi import APIRouter, HTTPException, Request
from __main import dbService

router = APIRouter()

@router.get("/api/messages")
async def getMessages(chatId: str):
    if(not chatId):
        raise HTTPException(status_code=400, detail="chatId is required!")
    
    try:
        messages = await dbService.getMessagesByChatId(chatId)
        return messages
    except Exception as err:
        raise HTTPException(status_code=500, detail="Failed to load messages")